boxes/generators/can_storage.py

```diff
@@ -0,0 +1,232 @@
+                    # draw barrier
+                    self.moveTo(0.5 * self.thickness, 1.1 * self.thickness + self.burn, 90)
+                    self.rectangularHole(0, 0, self.barrier_height, self.thickness, center_x=False, center_y=True, color=Color.ANNOTATIONS)
+
+            # bottom chute
+            with self.saved_context():
+                self.moveTo(0, 0.5 * self.thickness, self.angle)
+                self.fingerHolesAt(0, 0, self.depth / math.cos(math.radians(self.angle)), 0)
+            # top chute
+            with self.saved_context():
+                self.moveTo(0, self.thickness + self.canDiameter + self.bottom_chute_height + self.top_chute_height + 0.5 * self.thickness, -self.angle)
+                self.fingerHolesAt(0, 0, self.top_chute_depth, 0)
+            # front barrier
+            with self.saved_context():
+                self.moveTo(0.5 * self.thickness, 1.1 * self.thickness + self.burn, 90)
+                self.fingerHolesAt(0, 0, self.barrier_height, 0)
+            # fill with holes
+            border = [
+                (self.thickness, self.thickness), 
+                (self.depth, self.thickness + self.depth * math.sin(math.radians(self.angle))), 
+                (self.depth, self.height), 
+                (self.thickness + 0.75 * self.canDiameter, self.height),
+                (self.thickness + 0.75 * self.canDiameter, self.thickness + self.canDiameter + self.bottom_chute_height + self.top_chute_height + self.thickness - 0.75 * self.canDiameter * math.sin(math.radians(self.angle))),
+                (self.top_chute_depth, self.thickness + self.canDiameter + self.depth * math.sin(math.radians(self.angle)) + self.thickness),
+                (self.top_chute_depth, self.thickness + self.canDiameter + self.depth * math.sin(math.radians(self.angle))),
+                (self.thickness + 0.75 * self.canDiameter, self.thickness + self.canDiameter + self.bottom_chute_height + self.top_chute_height - 0.75 * self.canDiameter * math.sin(math.radians(self.angle))),
+                (self.thickness + 0.75 * self.canDiameter, self.thickness + self.bottom_chute_height ),
+                (self.thickness, self.thickness + self.bottom_chute_height ),
+                ]
+            if self.mydebug:
+                self.showBorderPoly(border)
+            if self.fillHoles_fill_pattern != "no fill":
+                self.fillHoles(
+                    pattern=self.fillHoles_fill_pattern, 
+                    border=border, 
+                    max_radius=self.fillHoles_hole_max_radius, 
+                    hspace=self.fillHoles_space_between_holes, 
+                    bspace=self.fillHoles_space_to_border, 
+                    min_radius=self.fillHoles_hole_min_radius, 
+                    style=self.fillHoles_hole_style,
+                    maxcnt=2000)            
+        
+    def render(self):
+        self.mydebug = False # set to True to enable debugging
```

Comment:
Please just use the regular debug parameter. It was added for exactly this purpose.

Reply:
--> done